packages/@react-aria/ssr/src/SSRProvider.tsx

```diff
@@ -75,7 +75,7 @@ export function useSSRSafeId(defaultId?: string): string {
     console.warn('When server rendering, you must wrap your application in an <SSRProvider> to ensure consistent ids are generated between the client and server.');
   }
 
-  return useMemo(() => defaultId || `react-aria-${ctx.prefix}-${++ctx.current}`, [defaultId]);
+  return useMemo(() => defaultId !== undefined ? defaultId : `react-aria-${ctx.prefix}-${++ctx.current}`, [defaultId]);
```

Comment:
what was this change for? shouldn't we allow `null`? empty string is also not a valid id... 🤔

Reply:
this came as a result of the changes to useSlotId https://github.com/adobe/react-spectrum/pull/2073/files#diff-452d47f1992f4562c6e4cc2218bee57897ad1fe96fbb6ff8aa952202170c5315R95

useSlotId was originally only running useId to get the default for the useState, then it managed its own instance of the id. This was getting disconnected from the merged id however, which useId returns
useSlotId could and did use null as a special indicator for when the id couldn't be found in the dom

because I changed useSlotId to no longer track its own id, useId will now try to merge including that special not found value, so I think this is the right way to allow that (I didn't break our SSR tests :) not that those are particularly great coverage for this stuff )